[docs] Add info about requirements for appVersion policy (#18621)
expo/expo: docs/pages/eas-update/runtime-versions.md

This runtime version policy is perfect if you are not including custom native code in your project and the only native changes you make are when upgrading Expo SDKs.

> ### `"appVersion"` runtime version policy (available starting from SDK 46)

Reviewer: ```suggestion
### `"appVersion"` runtime version policy (available in SDK 46 and higher)
```
